code/scripting/api/libs/mission.cpp

```diff
@@ -2177,6 +2178,44 @@ ADE_VIRTVAR(SkyboxAlpha, l_Mission, "number", "Sets or returns the current skybo
 	return ade_set_args(L, "f", Nmodel_alpha);
 }
 
+ADE_VIRTVAR(Skybox, l_Mission, "model", "Sets or returns the current skybox model", "model", "The skybox model")
+{
+	model_h* model = nullptr;
+	if (!ade_get_args(L, "*|o", l_Model.GetPtr(&model)))
+		return ade_set_error(L, "o", l_Model.Set(model_h()));
+
+	if (ADE_SETTING_VAR && model && model->isValid()) {
+		if (Nmodel_num >= 0) {
+			model_unload(Nmodel_num);
+			Nmodel_num = -1;
+		}
+
+		if (Nmodel_instance_num >= 0) {
+			model_delete_instance(Nmodel_instance_num);
+			Nmodel_instance_num = -1;
+		}
+
+		Nmodel_num = model->GetID();
+
+		if (Nmodel_num >= 0) {
+			model_page_in_textures(Nmodel_num);
+
+			Nmodel_instance_num = model_create_instance(model_objnum_special::OBJNUM_NONE, Nmodel_num);
+			The_mission.skybox_model_animations.initializeMoveables(model_get_instance(Nmodel_instance_num));
+		}
+
+		// Since we have a new skybox we need to rerender the environment map
+		stars_invalidate_environment_map();
```

Comment:
What about calling `stars_set_background_model` instead?  The `texture_name` parameter would be nullptr and the flags and alpha would be the current flags and alpha.

Reply:
That'd need some rework then.
texture_name nullptr would unload whatever was in Nbitmap before. Does that have unintended side-effects?
The function would also need to be changed to accept already loaded models over a model file path, as switching models has to work from preloaded files with scripting.

Reply:
Nbitmap is only meaningful in the context of the current model, so yes, unloading it when the model is changed is appropriate.  (In the original public source release, it is used as the spherec.pof texture file when the full nebula is shown.)

Changing the function to accept a model_num rather than a model_name looks like it could be done without much trouble, assuming a wrapper function that accepts model_name is also added.  The motivation here is to reduce code duplication in the script function.